### PR TITLE
invariant_booleans: Fix false positive in loops inside cases of break statements

### PR DESCRIPTION
**Pull request by alexeieleusis**
Fixes https://github.com/dart-lang/linter/issues/720

### CODE REVIEW

File: lib/src/util/condition_scope_visitor.dart [hunk 1]
```diff
@@ -255,6 +267,20 @@ abstract class ConditionScopeVisitor extends RecursiveAstVisitor {
   }
 
   @override
+  visitSwitchCase(SwitchCase node) {
+    node.visitChildren(this);
+    AstNode previousCase = node.parent.childEntities
+        .where((e) => e is AstNode)
+        .lastWhere((n) => n != node, orElse: () => null);
```
Comment (bwilkerson) [inline lib/src/util/condition_scope_visitor.dart]: It seems to me that
```
[0, 1, 2, 3, 4].lastWhere((e) => e != i)
```
will return `4` for every value of `i` other than `4`. I don't think this code works at all like you want it to, despite the test passing. (The test is incomplete; you need some cases that don't end in a break statement.)

File: lib/src/util/condition_scope_visitor.dart [hunk 2]
```diff
@@ -255,6 +267,20 @@ abstract class ConditionScopeVisitor extends RecursiveAstVisitor {
...
+    if (previousCase == null ||
+        previousCase.childEntities.last is! BreakStatement) {
```
Comment (bwilkerson) [inline lib/src/util/condition_scope_visitor.dart]: I think this also needs to account for return and throw statements, and all or some continue statements.

**Comment by pq** [comment]
Gentle ping: @alexeieleusis, is this one abandoned?

**Comment by alexeieleusis** [comment]
Apologies, I totally missed the previous ping. Right now I have a lot of things on my plate, I don't think I will be able to work on this in the next two weeks. I scheduled some time to look at this the first week of July, if I can't allocate enough time for it, I'll close the PR as abandoned then.

**Comment by pq** [comment]
Awesome and no sweat.  Thanks for the update!

**Comment by mit-mit** [comment]
Closing as assumed abandoned

**Pull request closed without merging**